src/api/migraphx.py

```diff
@@ -273,12 +273,25 @@ def operation(h):
                  fname='migraphx::load',
                  returns='migraphx::program')
 
+api.add_function('migraphx_load_buffer',
+                 api.params(buffer='const char*',
```

Comment:
I assume `buffer='const std::vector<char>&` doesnt work?

Reply:
No, if I use `buffer='const std::vector<char>&` , it gives an error while building.
```
+        *out = allocate<migraphx_program_t>(migraphx::load_buffer(
+            (const std::vector<char>&(buffer, buffer + buffer_size)), (options->object)));
```
There might be an error with the code generation since the '&' shouldn't be here in my opinion. 

If instead I were to use `buffer='const std::vector<char>` , it changes the  buffer argument to `char *` instead of `const char *` , which causes type conversion issues when trying to pass the buffer using `vector<char> .data()` method
[vector-diff.txt](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/files/9041496/vector-diff.txt)

Reply:
added fix.